gson/src/main/java/com/google/gson/FieldNamingPolicy.java

```diff
@@ -149,10 +150,6 @@ static String upperCaseFirstLetter(String name) {
       firstCharacter = name.charAt(++index);
     }
 
-    if (index == name.length()) {
```

Comment:
I’m anxious that this will break compatibility with code already deployed. At this point that’s much more important than actual correctness, which is both depressing and true.